framework_tvb/tvb/interfaces/rest/server/access_permissions/permissions.py

```diff
@@ -0,0 +1,84 @@
+# -*- coding: utf-8 -*-
+#
+#
+# TheVirtualBrain-Framework Package. This package holds all Data Management, and 
+# Web-UI helpful to run brain-simulations. To use it, you also need do download
+# TheVirtualBrain-Scientific Package (for simulators). See content of the
+# documentation-folder for more details. See also http://www.thevirtualbrain.org
+#
+# (c) 2012-2020, Baycrest Centre for Geriatric Care ("Baycrest") and others
+#
+# This program is free software: you can redistribute it and/or modify it under the
+# terms of the GNU General Public License as published by the Free Software Foundation,
+# either version 3 of the License, or (at your option) any later version.
+# This program is distributed in the hope that it will be useful, but WITHOUT ANY
+# WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS FOR A
+# PARTICULAR PURPOSE.  See the GNU General Public License for more details.
+# You should have received a copy of the GNU General Public License along with this
+# program.  If not, see <http://www.gnu.org/licenses/>.
+#
+#
+#   CITATION:
+# When using The Virtual Brain for scientific publications, please cite it as follows:
+#
+#   Paula Sanz Leon, Stuart A. Knock, M. Marmaduke Woodman, Lia Domide,
+#   Jochen Mersmann, Anthony R. McIntosh, Viktor Jirsa (2013)
+#       The Virtual Brain: a simulator of primate brain network dynamics.
+#   Frontiers in Neuroinformatics (7:10. doi: 10.3389/fninf.2013.00010)
+#
+#
+from abc import abstractmethod
+
+from tvb.core.entities.storage import CaseDAO, DatatypeDAO
+from tvb.core.services.exceptions import ProjectServiceException
+from tvb.core.services.project_service import ProjectService
+from tvb.interfaces.rest.server.request_helper import get_current_user
+
+
+class ResourceAccessPermission:
+    def __init__(self, resource_identifier):
+        self.resource_identifier = resource_identifier
+
+    def has_access(self):
+        current_user = get_current_user()
+        return current_user.id in self.get_resource_owners()
+
+    @abstractmethod
+    def get_resource_owners(self):
+        """
+        :return: a list of users id who can access the requested resource
+        """
+        raise RuntimeError("Not implemented.")
+
+
+class OperationAccessPermission(ResourceAccessPermission):
```

Comment:
This is a bit too restrictive.
A user able to access a Project, should be allowed to see all Operation under that project, as well as all DataTypes in the project.